scan/lib/scan/detect_values.rb

```diff
@@ -100,6 +109,63 @@ def self.filter_simulators(simulators, operator = :greater_than_or_equal, deploy
       end
     end
 
+    def self.default_os_version(os_type)
+      @os_versions ||= {}
+      @os_versions[os_type] ||= begin
+         UI.crash!("Unknown platform: #{os_type}") unless PLATFORMS.key?(os_type)
+         platform = PLATFORMS[os_type]
+
+         # list SDK version for currently running Xcode
+         sdks_output, = Open3.capture3('xcodebuild -showsdks -json')
+         sdk_version = begin
+           JSON.parse(sdks_output).find { |e| e['platform'] == platform[:simulator] }['sdkVersion']
+         rescue StandardError => e
+           UI.error(e)
+           UI.error("xcodebuild CLI broken, please run `xcodebuild` and make sure it works")
+           UI.user_error!("xcodebuild not working")
+         end
```

Comment:
Could be nice to capture the status as well, and check if `status.success?` in case the `xcodebuild` command exited with non-zero status code.

```suggestion
         sdk_version = begin
           sdks_output, _, status = Open3.capture3('xcodebuild -showsdks -json')
           raise status unless status.success?
           JSON.parse(sdks_output).find { |e| e['platform'] == platform[:simulator] }['sdkVersion']
         rescue StandardError => e
           UI.error(e)
           UI.error("xcodebuild CLI broken, please run `xcodebuild` and make sure it works")
           UI.user_error!("xcodebuild not working")
         end
```